homeassistant/components/tellduslive.py

```diff
@@ -0,0 +1,139 @@
+CONF_TOKEN = "token"
+CONF_TOKEN_SECRET = "token_secret"
+
+REQUIREMENTS = ['tellive-py==0.5.2']
+_LOGGER = logging.getLogger(__name__)
+
+NETWORK = None
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=600)
+
+
+class TelldusLiveData(object):
+    """ Gets the latest data and update the states. """
+
+    def __init__(self, config):
+        public_key = config[DOMAIN].get(CONF_PUBLIC_KEY)
+        private_key = config[DOMAIN].get(CONF_PRIVATE_KEY)
+        token = config[DOMAIN].get(CONF_TOKEN)
+        token_secret = config[DOMAIN].get(CONF_TOKEN_SECRET)
+
+        from tellive.client import LiveClient
+        self.sensors = []
+        self.client = LiveClient(public_key=public_key,
+                                 private_key=private_key,
+                                 access_token=token,
+                                 access_secret=token_secret)
+
+    def request(self, what, params=None):
+        """ Sends a request to the tellstick live API """
+        supported_methods = TELLSTICK_TURNON \
+            | TELLSTICK_TURNOFF \
+            | TELLSTICK_TOGGLE
+        params = params or dict()
+        params.update({"supportedMethods": supported_methods,
+                       'extras': 'coordinate,timezone,tzoffset'})
+        return self.client.request(what, params)
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def _update_sensors(self):
+        """ Get the latest data from Telldus Live """
+        _LOGGER.info("Updating sensors for real")
+        params = {"includeValues": 1,
+                  "includeScale": 1}
+        response = self.request("sensors/list", params)
+        return response["sensor"]
+
+    def get_sensors(self):
+        """ Get the latest (possibly cached) sensor values """
+        updated_sensors = self._update_sensors()
+        if updated_sensors is not None:
+            _LOGGER.info("tellduslive data updated successfully.")
+            self.sensors = updated_sensors
+        return self.sensors
+
+    def get_switches(self):
+        """ Get the configured switches """
+        response = self.request("devices/list")
+        return response["device"]
+
+
+def setup(hass, config):
+    """ Setup the tellduslive component """
+
+    # later: aquire an app key and authenticate using username + password
+    if not validate_config(config, {DOMAIN: [CONF_PUBLIC_KEY,
+                                             CONF_PRIVATE_KEY,
+                                             CONF_TOKEN,
+                                             CONF_TOKEN_SECRET]}, _LOGGER):
+        _LOGGER.error(
+            "Configuration Error: "
+            "Please make sure you have configured your keys "
+            "that can be aquired from https://api.telldus.com/keys/index")
+        return False
+
+    global NETWORK
+    NETWORK = TelldusLiveData(config)
+
+    for component_name, discovery_type in (
+            ('switch', DISCOVER_SWITCHES),
```

Comment:
Can you make sure that you only fire the discovery event if there are actually sensors available?